staging/src/k8s.io/apiserver/pkg/server/genericapiserver.go

```diff
@@ -272,6 +276,10 @@ func (s preparedGenericAPIServer) NonBlockingRun(stopCh <-chan struct{}) error {
 		close(internalStopCh)
 	}()
 
+	if s.AuditBackend != nil {
+		go s.AuditBackend.Run(stopCh)
```

Comment:
I have put it here because we check liveness before running the hook, don't we? But if auditing isn't up, this might block? Or maybe I mix that up with OpenShift.